src/cmdstan/stansummary.cpp

```diff
@@ -36,131 +36,68 @@ Example:  stansummary model_chain_1.csv model_chain_2.csv
   -s, --sig_figs [n]          Significant figures reported. Default is 2.
                               Must be an integer from (1, 18), inclusive.
 )";
-  if (argc < 2) {  // pre-empt boost::program_options
+  if (argc < 2) {
     std::cout << usage << std::endl;
     return -1;
   }
 
   // Command-line arguments
-  int sig_figs;
+  int sig_figs = 2;
   int autocorr_idx;
   std::string csv_filename;
-  std::string percentiles_spec;
+  std::string percentiles_spec = "5,50,95";
   std::vector<std::string> filenames;
-  boost::program_options::options_description desc("Allowed options");
-  desc.add_options()("help,h", "Produce help message")(
-      "sig_figs,s",
-      boost::program_options::value<int>(&sig_figs)->default_value(2),
-      "Significant figures, default 2.")(
-      "autocorr,a", boost::program_options::value<int>(&autocorr_idx),
-      "Display the chain autocorrelation.")(
-      "csv_filename,c",
-      boost::program_options::value<std::string>(&csv_filename),
-      "Write statistics to a csv.")(
-      "percentiles,p",
-      boost::program_options::value<std::string>(&percentiles_spec)
-          ->default_value("5,50,95"),
-      "Percentiles to report.")(
-      "input_files,i",
-      boost::program_options::value<std::vector<std::string> >(&filenames),
-      "Sampler csv files. ");
-  boost::program_options::positional_options_description p;
-  p.add("input_files", -1);
 
-  // Parse, validate command-line
-  boost::program_options::variables_map vm;
+  CLI::App app{"Allowed options"};
+  app.add_option("--sig_figs,-s", sig_figs, "Significant figures, default 2.",
+                 true)
+      ->check(CLI::Range(1, 18));
+  app.add_option("--autocorr,-a", autocorr_idx,
+                 "Display the chain autocorrelation.", true)
+      ->check(CLI::PositiveNumber);
+  app.add_option("--csv_filename,-c", csv_filename,
+                 "Write statistics to a csv.", true)
+      ->check(CLI::NonexistentPath);
```

Comment:
This checks that the output file is in a writeable directory?

Reply:
line 68 just checks that you're not clobbering an existing file. 
lines 92-100 check that the file is writeable.

looks like C++17 will have package "filesystem" that will do a nicer job on these checks.